### PR TITLE
Fix audio related issues

### LINKED ISSUE #275: Unable to mute the Preview in the Editor
Seems to be some sort of regression. In the past I believe you'd only be able to mute files that were captured with audio for formats that support it, but in the new Editor you should always be able to mute if you had audio recording on when capturing.

### LINKED ISSUE #692: DOMException: The play() request was interrupted
Sentry issue: [KAP-3CG](https://sentry.io/organizations/wulkano-l0/issues/699070897/?referrer=github_integration)

https://developers.google.com/web/updates/2017/06/play-request-was-interrupted

### LINKED ISSUE #693: Audio not being recorded in the new beta 3.6
macOS version:        10.13.6
Kap version:          BETA 3.6

Hello, i have download the new beta version and of course deleted the file ~/Library/Application Support/Kap Beta/config.json

But no audio from my mic is recorded, have tried everything, permissions are granted, audio is enabled, so i unistalled the new beta version and installed back the 3.5 and everything is working again.

### PR DESCRIPTION
**Pull request by karaggeorge**
Fixes #692 
Fixes #693 
Fixes #275 

This also adds a lot of checks around having/granting access to the microphone, which might help reduce or troubleshoot problems people are having with recording sound

Also, the last beta was recording sound for me, so if someone can validate that this fixes #693 (using the build) that would be great

### CODE REVIEW

File: main/cropper.js [hunk 1]
```diff
@@ -79,9 +80,40 @@ const openCropper = (display, activeDisplayId) => {
...
+  const recordAudio = settings.get('recordAudio');
+
+  if (recordAudio && !hasMicrophoneAccess()) {
+    const granted = await ensureMicrophonePermissions(async () => {
+      const {response} = await dialog.showMessageBox({
+        type: 'warning',
+        buttons: ['System Preferences', 'Continue'],
+        defaultId: 1,
+        message: 'Can not access audio input device',
+        detail: 'Recoding audio is enabled but Kap does not have access to the microphone. Continue without audio or grant microphone access to Kap in your system preferences.',
```
Comment (sindresorhus) [inline main/cropper.js]:
```suggestion
        detail: 'Audio recording is enabled but Kap does not have access to the microphone. Continue without audio or grant Kap access to the microphone the System Preferences.',
```

File: package.json [hunk 2]
```diff
@@ -124,6 +124,7 @@
       "minimumSystemVersion": "10.12.0",
       "darkModeSupport": true,
       "extendInfo": {
+        "NSMicrophoneUsageDescription": "Enable sound recording by giving Kap access to your audio input devices.",
```
Comment (sindresorhus) [inline package.json]:
```suggestion
        "NSMicrophoneUsageDescription": "Kap needs access to the microphone to be able to record audio for screen recordings.",
```

**Comment by sindresorhus** [comment]
@Geroimparable @kevincannon Can you test https://kap-artifacts.now.sh/fix-recording-with-sound and see if it fixes #693 for you?

**Comment by kevincannon** [comment]
Just teested this and it worked for me! 💪💪

**Pull request merged**